### ISSUE TITLE
[BUG] Config Option: checkItemDurability Not Working

### ISSUE THREAD

**Issue by Relaxing9** [snowgears/shopbugs#235]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Type of bug

Other unexpected behaviour

### Environment

```markdown
- Server Version: Paper 1.17.1 (Build 100)
- Shop Version: 1.8.2.7.5
```

### Server startup log

https://paste.gg/p/anonymous/6f74c5c5546946f2a2e3c2ca58ba31f2

### Shop config files

https://paste.gg/p/anonymous/44d6dfc7a644467c97fa8676331589ad

### Bug description

Items that have loss of durability are not being sold on shops, I have also heard that placing some items in a shop, has the possibility of the shop repairing a damaged item on a server restart.

![image](https://user-images.githubusercontent.com/41935047/125368440-3918a900-e348-11eb-87f9-f12c7f454b6d.png)

### Steps to reproduce

1. Create a shop
2. Try to sell Tridents with different disabilities
3. Observe

### Expected behaviour

All Tridents should be able to be sold, not just a Trident that matches the initial ones durability.

### Actual behaviour

Shops claim there is no stock of an item when there is (with different durabilites)

![2021-07-12_17 42 04](https://user-images.githubusercontent.com/41935047/125368147-a24bec80-e347-11eb-9c45-b5da3d95ec56.png)
![2021-07-12_17 42 02](https://user-images.githubusercontent.com/41935047/125368156-a546dd00-e347-11eb-8cea-6cf232e31aa2.png)

### Other information

`# This will make it so shops will only complete transactions if the item durabilities are the same #
checkItemDurability: false`

**Comment by Relaxing9** [comment]
![image](https://user-images.githubusercontent.com/41935047/125630295-419a14f3-b27c-46cb-9dd7-ea5ea83bf8da.png)

Related to this issue.

**Comment by snowgears** [comment]
Thanks for the additional info!

**Comment by snowgears** [comment]
Fixed in 1.8.2.8 (in progress)
